Content.Shared/MobState/EntitySystems/MobStateSystem.cs

```diff
@@ -37,6 +40,7 @@ public override void Initialize()
             SubscribeLocalEvent<MobStateComponent, DamageChangedEvent>(UpdateState);
             SubscribeLocalEvent<MobStateComponent, UpdateCanMoveEvent>(OnMoveAttempt);
             SubscribeLocalEvent<MobStateComponent, StandAttemptEvent>(OnStandAttempt);
+            SubscribeLocalEvent<MobStateComponent, MobStateChangedEvent>(OnMobStateChanged);
```

Comment:
you're in the system where this is relevant anyway, you shouldn't subscribe to its own event. just check where the code for changing to the 'dead' state is and add the popup there